### PR TITLE
Rework the prerelease-check task to use workspaces

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Stop using pipeline resources, use workspaces instead.
For the gcs object, use gsutils directly to stat the object,
which saves us downloading the whole bucket to check a single file.

Signed-off-by: Andrea Frittoli <andrea.frittoli@uk.ibm.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

/kind misc

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
/meow

**Comment by afrittoli** [comment]
@tektoncd/plumbing-maintainers can I get a second pair of eyes on this one please? 👀

**Pull request merged**
